lib/config.js

```diff
@@ -140,8 +150,35 @@ function getLocalConfig(thisConfig, directory) {
         config = ConfigOps.merge(localConfig, config);
     }
 
-    // Use the personal config file if there are no other local config files found.
-    return found || thisConfig.useSpecificConfig ? config : ConfigOps.merge(config, getPersonalConfig());
+    if (!found && !thisConfig.useSpecificConfig) {
+
+        /*
+         * - Is there a personal config in the user's home directory? If so,
+         *   merge that with the passed-in config.
+         * - Otherwise, throw an exception.
+         */
+        personalConfig = getPersonalConfig();
+
+        if (personalConfig) {
+            config = ConfigOps.merge(config, personalConfig);
+        }
+    }
+
+    if (!found && !thisConfig.useSpecificConfig && !personalConfig && !optionsHaveRules(thisConfig.options)) {
```

Comment:
Hmm. I can see that. I'm leaning towards extracting to a few more functions. See my latest comment on the issue itself for how I plan to handle this.